pkg/driverbuilder/builder/ubuntu.go

```diff
@@ -3,111 +3,71 @@ package builder
+	ModuleFullPath       string
+	BuildProbe           bool
+	BuildModule          bool
+	GCCVersion           string
 }
 
 // Script compiles the script to build the kernel module and/or the eBPF probe.
-func (v ubuntuGeneric) Script(c Config) (string, error) {
-	t := template.New(string(TargetTypeUbuntuGeneric))
+func (v ubuntu) Script(c Config) (string, error) {
+
+	t := template.New(string(TargetTypeUbuntu))
+
 	parsed, err := t.Parse(ubuntuTemplate)
 	if err != nil {
 		return "", err
 	}
 
+	// read in the build config
 	kr := kernelReleaseFromBuildConfig(c.Build)
+	fmt.Printf("%+v\n", kr)
 
 	var urls []string
-    if c.KernelUrls == nil {
-        urls, err = ubuntuGenericHeadersURLFromRelease(kr, c.Build.KernelVersion)
-    } else {
-        urls, err = getResolvingURLs(c.KernelUrls)
-    }
-    if err != nil {
-        return "", err
-    }
-    if len(urls) < 2 {
-        return "", fmt.Errorf("specific kernel headers not found")
-    }
-
-	td := ubuntuTemplateData{
-		DriverBuildDir:       DriverDirectory,
-		ModuleDownloadURL:    fmt.Sprintf("%s/%s.tar.gz", c.DownloadBaseURL, c.Build.DriverVersion),
-		KernelDownloadURLS:   urls,
-		KernelLocalVersion:   kr.FullExtraversion,
-		KernelHeadersPattern: "linux-headers*generic",
-		ModuleDriverName:     c.Build.ModuleDriverName,
-		ModuleFullPath:       ModuleFullPath,
-		BuildModule:          len(c.Build.ModuleFilePath) > 0,
-		BuildProbe:           len(c.Build.ProbeFilePath) > 0,
-		GCCVersion:           ubuntuGCCVersionFromKernelRelease(kr),
+	if c.KernelUrls == nil {
+		urls, err = ubuntuHeadersURLFromRelease(kr, c.Build.KernelVersion)
+	} else {
+		urls, err = getResolvingURLs(c.KernelUrls)
 	}
-
-	if kr.IsGKE() {
-		td.KernelHeadersPattern = "linux-headers*gke"
-	}
-
-	buf := bytes.NewBuffer(nil)
-	err = parsed.Execute(buf, td)
+	// if there was an error
 	if err != nil {
 		return "", err
 	}
-	return buf.String(), nil
-}
-
-// ubuntuAWS is a driverkit target.
-type ubuntuAWS struct {
-}
-
-// Script compiles the script to build the kernel module and/or the eBPF probe.
-func (v ubuntuAWS) Script(c Config) (string, error) {
-	t := template.New(string(TargetTypeUbuntuGeneric))
-	parsed, err := t.Parse(ubuntuTemplate)
-	if err != nil {
-		return "", err
+	// if no results
+	if len(urls) == 0 {
+		return "", fmt.Errorf("specific kernel headers not found")
 	}
 
-	kr := kernelReleaseFromBuildConfig(c.Build)
-
-	var urls []string
-    if c.KernelUrls == nil {
-        urls, err = ubuntuAWSHeadersURLFromRelease(kr, c.KernelVersion)
-    } else {
-        urls, err = getResolvingURLs(c.KernelUrls)
-    }
-    if err != nil {
-        return "", err
-    }
-    if len(urls) < 2 {
-        return "", fmt.Errorf("specific kernel headers not found")
-    }
-
 	td := ubuntuTemplateData{
 		DriverBuildDir:       DriverDirectory,
-		ModuleDownloadURL:    moduleDownloadURL(c),
+		ModuleDownloadURL:    fmt.Sprintf("%s/%s.tar.gz", c.DownloadBaseURL, c.Build.DriverVersion),
```

Comment:
Isn't this the exact definition of `moduleDownloadURL`?
https://github.com/falcosecurity/driverkit/blob/76daf8787f96e91092cf8b16ba07c12d39173677/pkg/driverbuilder/builder/builders.go#L51
I'd leave old line here.

Reply:
Yea, I had actually not made this change. It was pulled in from somewhere else. I do need to flip it back.